### DISCUSSION TITLE
how to make custom hooks in react?

### DISCUSSION THREAD

**Discussion by larrybrowndev** [drknzz/GitHub-Achievements#298] [Q&A] [answered]
I'm making a chatting react project and need to make custom hooks, help me!

**Comment by topdev1228** [comment, marked as answer]
Hi here is my solution for that:

function useCustomHook() {
  const [count, setCount] = React.useState(0);

  const increment = () => {
    setCount((prevCount) => prevCount + 1);
  };

  return { count, increment };
}

import React from 'react';
import useCustomHook from './useCustomHook';

function MyComponent() {
  const { count, increment } = useCustomHook();

  return (
    <div>
      <p>Count: {count}</p>
      <button onClick={increment}>Increment</button>
    </div>
  );
}

export default MyComponent;

**Reply by larrybrowndev** [reply]
Great, thanks!
